qiskit/providers/fake_provider/fake_backend_v2.py

```diff
@@ -42,7 +42,7 @@ def __init__(self):
             None,
             name="FakeV2",
             description="A fake BackendV2 example",
-            online_date=datetime.datetime.utcnow(),
+            online_date=datetime.datetime.now(datetime.timezone.utc),
```

Comment:
I was confused why I hadn't seen and fixed this when I was doing the initial Python 3.12 testing, but then I remembered that symengine didn't have wheels at the time and we had Numpy/Scipy problems, so I'd not got all the way.